kfs-core/src/main/java/edu/arizona/kfs/gl/batch/service/impl/BudgetAdjustmentCashTransferServiceImpl.java

```diff
@@ -0,0 +1,596 @@
+              	   SystemOptions options = getOptionsService().getOptions(e.getUniversityFiscalYear());
+                   if (ObjectUtils.isNull(options)) {
+                       options = getOptionsService().getCurrentYearOptions();
+                   }
+                                      
+                   e.setFinancialBalanceTypeCode(options.getActualFinancialBalanceTypeCd());
+                   e.setFinancialObjectTypeCode(options.getFinancialObjectTypeTransferIncomeCd());
+                   e.setTransactionLedgerEntryDescription(ba.getTransactionLedgerEntryDescription());
+                   e.setTransactionDate(new java.sql.Date(runDate.getTime())); 
+                   e.setOrganizationDocumentNumber(ba.getOrganizationDocumentNumber());
+                   e.setProjectCode(ba.getProjectCode()); 
+                   
+                   // set fiscal period, if next fiscal year set to 01, else leave to current period
+                   if (currentFiscalYear.equals(e.getUniversityFiscalYear() - 1)) {
+                       e.setUniversityFiscalPeriodCode(BudgetAdjustmentDocumentRuleConstants.MONTH_1_PERIOD_CODE);
+                   }
+
+                   try {
+                       createOutputEntry(e, group);
+                       reportOriginEntryGenerated++;                                                              
+                   }
+                   catch (IOException ioe) {
+                       LOG.error("generateBudgetAdjustmentCashTransferTransactions Stopped: " + ioe.getMessage());
+                       throw new RuntimeException("generateBudgetAdjustmentCashTransferTransactions Stopped: " + ioe.getMessage(), ioe);
+                   }                   
+                   
+                } //matching transaction found 
+            }
+            catch (RuntimeException re) {
+                LOG.error("generateBudgetAdjustmentCashTransferTransactions Stopped: " + re.getMessage());
+                throw new RuntimeException("generateBudgetAdjustmentCashTransferTransactions Stopped: " + re.getMessage(), re);
+            }
+            catch (Exception ex) {
+                List errorList = new ArrayList();
+                errorList.add(new Message(ex.toString() + " occurred for this record.", Message.TYPE_FATAL));
+                reportWriterService.writeError(ba, errorList);
+            }
+        }
+        return reportOriginEntryGenerated;
+    }
+        
+    protected void createOutputEntry(OriginEntryInformation oef, PrintStream group) throws IOException {       
+        try {
+            group.printf("%s\n", oef.getLine());
+        }
+        catch (Exception e) {
+            throw new IOException(e.toString());
+        }
+    }
+    
+    protected void createOutputEntry(String line, PrintStream ps) throws IOException {
+        try {
+            ps.printf("%s\n", line);
+        } catch (Exception e) {
+            throw new IOException(e.toString());
```

Comment:
Constructor for new exceptions take a Throwable as argument, if you don't intend to add more info in the message.